subiquity/server/server.py

```diff
@@ -663,7 +663,38 @@ def validate_autoinstall(self):
                 context=ctx,
             )
 
-    def load_autoinstall_config(self, *, only_early):
+    @with_context(name="read_config")
+    def _read_config(self, *, cfg_path: str, context: Context) -> dict[str, Any]:
+        with open(cfg_path) as fp:
+            config: dict[str, Any] = yaml.safe_load(fp)
+
+        autoinstall_config: dict[str, Any] = dict()
```

Comment:
```suggestion
        autoinstall_config: dict[str, Any] = None
```
All code paths set `autoinstall_config`, so we can skip the unused `dict()`

Reply:
Fair, although really I was just trying to annotate the type once and avoid an `| None`. What if we just do 

 `autoinstall_config: dict[str, Any]`

and defer assignment?

Reply:
works for me